pyhive/sqlalchemy_hive.py

```diff
@@ -517,6 +517,22 @@ def _check_unicode_description(self, connection):
         # We decode everything as UTF-8
         return True
 
+
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import Insert
+
+
+@compiles(Insert, 'hive')
+def insert_into_add_table_keyword(insert, compiler, **kwargs):
```

Comment:
I think a simpler way is to override visit_insert in HiveCompiler. The overridden method would call the base method, assert that the result starts with INSERT INTO, and swap that out with INSERT INTO TABLE.

Also could you throw in a test? Something that makes a test table (default db is fine), inserts a couple rows into it, and reads the results back.